front/src/modules/ui/board/components/BoardColumn.tsx

```diff
@@ -52,49 +61,56 @@ const StyledAmount = styled.div`
 `;
 
 type OwnProps = {
-  colorCode?: string;
+  color?: string;
   title: string;
   pipelineStageId?: string;
   onTitleEdit: (title: string) => void;
+  onColumnColorEdit: (color: string) => void;
   totalAmount?: number;
   children: React.ReactNode;
   isFirstColumn: boolean;
 };
 
 export function BoardColumn({
-  colorCode,
+  color,
   title,
   onTitleEdit,
+  onColumnColorEdit,
   totalAmount,
   children,
   isFirstColumn,
 }: OwnProps) {
-  const [isEditing, setIsEditing] = React.useState(false);
-  const [internalValue, setInternalValue] = React.useState(title);
+  const [isBoardColumnMenuOpen, setIsBoardColumnMenuOpen] =
+    React.useState(false);
 
-  const debouncedOnUpdate = debounce(onTitleEdit, 200);
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setInternalValue(event.target.value);
-    debouncedOnUpdate(event.target.value);
-  };
+  const setHotkeyScope = useSetHotkeyScope();
+  setHotkeyScope(BoardColumnHotkeyScope.BoardColumn, { goto: false });
+  useScopedHotkeys(
+    [Key.Escape, Key.Enter],
+    () => setIsBoardColumnMenuOpen(false),
+    BoardColumnHotkeyScope.BoardColumn,
+    [],
+  );
 
   return (
     <StyledColumn isFirstColumn={isFirstColumn}>
-      <StyledHeader onClick={() => setIsEditing(true)}>
-        <StyledColumnTitle color={colorCode}>
-          {isEditing ? (
-            <EditColumnTitleInput
-              color={colorCode}
-              onFocusLeave={() => setIsEditing(false)}
-              value={internalValue}
-              onChange={handleChange}
-            />
-          ) : (
-            <div>{title}</div>
-          )}
-        </StyledColumnTitle>
+      <StyledHeader>
+        <Tag
+          onClick={() => setIsBoardColumnMenuOpen(true)}
```

Comment:
Put this in a handler and call setHotkeyScope in it. 

Because otherwise every column will set hotkeyscope at each rerender and we want to maximize setting hotkey scope in an imperative way. So usually the best place to do it is in handlers.